### PR TITLE
JDK-8325908: Finish removal of IntlTest and CollatorTest

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this PR which fixes / finishes the rest of the IntlTest test framework removal in java.text and java.util.i18n tests.

For context, the IntlTest class only ran methods prefixed by _test_ or _Test_ with public visibility and was originally removed due to some tests spuriously passing that were not aware of the specific running requirements.

This PR includes the following changes,
- There were some tests with package-private visibility that were never ran by IntlTest. Those tests do appear to be valid tests, and thus have been updated with `@Test` annotation.
- The test method DFSExponenTest() in DFSExponential.java was not prefixed by test and thus never ran by IntlTest. It is a valid test and should be ran as well.
- DateFormatRoundTripTest.java was supposed to remain a non JUnit test, however the run invocation was converted to `@run junit ...` this has been switched back to `@run main ...`
- There were two instances of the script missing some tests that should have had their methods updated with the `@Test` annotation: APITest.java and DFSSerialization.java. These tests have been updated accordingly.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8325908](https://bugs.openjdk.org/browse/JDK-8325908): Finish removal of IntlTest and CollatorTest (**Bug** - P4)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [c04309a7](https://git.openjdk.org/jdk/pull/17858/files/c04309a73b6c9162e41b519608a6dda84d1a3ce4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17858/head:pull/17858` \
`$ git checkout pull/17858`

Update a local copy of the PR: \
`$ git checkout pull/17858` \
`$ git pull https://git.openjdk.org/jdk.git pull/17858/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17858`

View PR using the GUI difftool: \
`$ git pr show -t 17858`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17858.diff">https://git.openjdk.org/jdk/pull/17858.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17858#issuecomment-1944956040)

### CODE REVIEW

File: test/jdk/java/text/Format/NumberFormat/DFSExponential.java [hunk 1]
```diff
@@ -30,18 +30,18 @@
...
 import static org.junit.jupiter.api.Assertions.fail;
 
-public class DFSExponential
-{
-
+public class DFSExponential {
 
-   public void DFSExponenTest() throws Exception {
+    @Test
+    public void DFSExponenTest() throws Exception {
```
Comment (naotoj) [inline test/jdk/java/text/Format/NumberFormat/DFSExponential.java:44]: Not your change, but the method name reads odd to me. `TestDFSExponential` may be better alined with the other test for serialization.

**Review by naotoj — APPROVED** [review]
LGTM

File: test/jdk/java/text/Format/NumberFormat/DFSExponential.java (hunk 1 above)
Comment (justin-curtis-lu) [reply test/jdk/java/text/Format/NumberFormat/DFSExponential.java]: Done, thanks for the review.

**Pull request closed without merging**